astroquery/hsa/tests/test_hsa_remote.py

```diff
@@ -0,0 +1,117 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import pytest
+
+import os
+
+from ..core import HSAClass
+from .test_hsa import TestHSA
+
+
+class TestHSARemote(TestHSA):
+
+    @pytest.mark.remote_data
+    def test_download_data_observation_pacs(self):
+        obs_id = "1342195355"
+        parameters = {'retrieval_type': "OBSERVATION",
+                      'observation_id': obs_id,
+                      'instrument_name': "PACS"}
+        expected_res = obs_id + ".tar"
+        hsa = HSAClass(self.get_dummy_tap_handler())
+        res = hsa.download_data(**parameters)
+        assert res == expected_res
+        assert os.path.isfile(res)
+        os.remove(res)
```

Comment:
Please use a temp_dir type pytest fixture, as these files are not cleaned up in case there is a test failure. There are examples for this approach is a few other modules, e.g in esasky

Reply:
I'm using `tempfile.TemporaryDirectory()` now and I've added a `download_dir` parameter to all the functions.